### PR TITLE
 fix Path and ArrayPath types for defined value type with interface 

### PR DESCRIPTION
**Pull request by kotarella1110**
ref: https://github.com/react-hook-form/react-hook-form/pull/6605

### CODE REVIEW

**Comment by bluebill1049** [comment]
build pass 👁️

File: src/types/utils.ts [hunk 1]
```diff
@@ -73,13 +75,15 @@ export type Path<T> = T extends ReadonlyArray<infer V>
...
-type ArrayPathImpl<K extends string | number, V> = V extends Primitive
-  ? never
+type ArrayPathImpl<K extends string | number, V> = IsAny<V> extends true
+  ? `${K}` | `${K}.${ArrayPath<V>}`
   : V extends ReadonlyArray<infer U>
-  ? U extends Primitive
-    ? never
-    : `${K}` | `${K}.${ArrayPath<V>}`
-  : `${K}.${ArrayPath<V>}`;
+  ? U extends ReadonlyArray<any> | Record<any, unknown>
```
Comment (bluebill1049) [inline src/types/utils.ts:81]: I guess this is the part checking if it's `interface`? any chance we could just add a smile unit test? with interface usage, I can help if you are busy as well.

**Review by bluebill1049 — APPROVED** [review]
LGTM! Thank you @kotarella1110

File: src/types/utils.ts (hunk 1 above)
Comment (kotarella1110) [reply src/types/utils.ts:81]: I will add unit test!

**Comment by bluebill1049** [comment]
this pr is related to this right: https://github.com/react-hook-form/react-hook-form/issues/6693

**Comment by bluebill1049** [comment]
let's merge this in, we can follow up with the test later on.

**Pull request merged**